frontends/p4/uniqueNames.cpp

```diff
@@ -97,9 +97,9 @@ UniqueParameters::UniqueParameters(ReferenceMap *refMap, TypeMap *typeMap)
 
 IR::ID *RenameSymbols::getName() const {
     auto orig = getOriginal<IR::IDeclaration>();
-    if (!renameMap->toRename(orig)) return nullptr;
-    auto newName = renameMap->getName(orig);
-    auto name = new IR::ID(orig->getName().srcInfo, newName, orig->getName().originalName);
+    auto newName = renameMap->get(orig);
+    if (!newName.has_value()) return nullptr;
+    auto name = new IR::ID(orig->getName().srcInfo, *newName, orig->getName().originalName);
```

Comment:
Tangential comment but what if `originalName` is `nullptr`? Then the name is lost. 

Maybe this should call `toString`?

Reply:
`originalName` should always be set for `ID`s that correspond to things in the source code -- `originalName == nullptr` implies that this is a purely compiler generated symbol with no "original" name.  For these purely synthetic names, preserving whatever was first generated for them is probably not useful?